v2/internal/controllers/crd_aks_fleet_20230315preview_test.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	aks "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230202preview"
+	fleet "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230315preview"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	. "github.com/onsi/gomega"
+)
+
+func Test_AKS_Fleet_20230315_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateTestResourceGroupAndWait()
+	region := to.Ptr("westus3")
+	flt := &fleet.Fleet{
+		ObjectMeta: tc.MakeObjectMeta("fleet"),
+		Spec: fleet.Fleet_Spec{
+			Location: region,
+			Owner:    testcommon.AsOwner(rg),
+			HubProfile: &fleet.FleetHubProfile{
+				DnsPrefix: to.Ptr("aso"),
+			},
+			Tags: map[string]string{
+				"name": "test-tag",
+			},
+		},
+	}
+	// creating a fleet
+	tc.CreateResourceAndWait(flt)
+	tc.Expect(flt.Status.Id).ToNot(BeNil())
+	tc.Expect(flt.Status.Tags).ToNot(BeNil())
+	tc.Expect(flt.Spec.Tags["name"]).To(Equal("test-tag"))
+	armId := *flt.Status.Id
+
+	// patching a fleet
+	old := flt.DeepCopy()
+	flt.Spec.Tags = map[string]string{
+		"name": "test-tag2",
+	}
+	tc.PatchResourceAndWait(old, flt)
+	tc.Expect(flt.Spec.Tags["name"]).To(Equal("test-tag2"))
+
+	adminUsername := "adminUser"
+	sshPublicKey, err := tc.GenerateSSHKey(2048)
+	tc.Expect(err).ToNot(HaveOccurred())
+
+	identityKind := aks.ManagedClusterIdentity_Type_SystemAssigned
+	osType := aks.OSType_Linux
+	agentPoolMode := aks.AgentPoolMode_System
+	cluster := &aks.ManagedCluster{
+		ObjectMeta: tc.MakeObjectMeta("mc"),
+		Spec: aks.ManagedCluster_Spec{
+			Location:  region,
+			Owner:     testcommon.AsOwner(rg),
+			DnsPrefix: to.Ptr("aso"),
+			AgentPoolProfiles: []aks.ManagedClusterAgentPoolProfile{
+				{
+					Name:   to.Ptr("ap1"),
+					Count:  to.Ptr(1),
+					VmSize: to.Ptr("Standard_DS2_v2"),
+					OsType: &osType,
+					Mode:   &agentPoolMode,
+				},
+			},
+			LinuxProfile: &aks.ContainerServiceLinuxProfile{
+				AdminUsername: &adminUsername,
+				Ssh: &aks.ContainerServiceSshConfiguration{
+					PublicKeys: []aks.ContainerServiceSshPublicKey{
+						{
+							KeyData: sshPublicKey,
+						},
+					},
+				},
+			},
+			Identity: &aks.ManagedClusterIdentity{
+				Type: &identityKind,
```

Comment:
It looks like you're only using `identityKind` here - you could write `to.Ptr(aks.ManagedClusterIdentity_Type_SystemAssigned)` directly if you chose.